### ISSUE TITLE
feat(connector-corda): support corda 5 RC via TS/HTTP (no JVM)

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cacti#1480]
### Description

- Currently the Corda 4.x support in Cactus comes in the form of a JVM app under `packages/cactus-plugin-ledger-connector-corda/src/main-server`
- We need Corda 5 support which can now be implemented in pure TS/JS (no more forced JVM, yaay) under `packages/cactus-plugin-ledger-connector-corda/src/main/typescript`
- For now contract deployment is a lower priority task and do not need to be implemented in the scope of this GH issue
- On a high level, most of the work will (probably) be done in the `PluginLedgerConnectorCorda` class which needs the constructor argument (config type) extended with connectivity details about the Corda 5 ledger that it is going to be talking to. Once those connection details are available the connector can instantiate an axios instance and then use that to talk to the ledger the same way that the Corda 5 AIO image's README showcases flow triggering via HTTP from the command line with `curl`. 

### Relevant discussion

Make sure to read this relevant discussion on this PR about where to put the code that uses the new Corda 5 TS/HTTP communication:
https://github.com/hyperledger/cactus/pull/1347

### Useful Links

https://docs.r3.com/en/platform/corda/5.0-dev-preview-1.html
https://www.corda.net/blog/corda-5-developer-preview-running-your-first-cordapp/
https://www.corda.net/blog/corda-5-developer-preview-modular-apis/

cc: @elenaizaguirre

**Comment by ruzell22** [comment]
Hello @petermetz , can you assign me on this one? Thank you.

**Comment by petermetz** [comment]
> Hello @petermetz , can you assign me on this one? Thank you.

@ruzell22 Done, thank you!

**Comment by jagpreetsinghsasan** [comment]
Hi @ruzell22 
Thankyou for picking up the issue.
Do you have any update on this (or need some help/guidance in doing this?)

**Comment by ruzell22** [comment]
Hello @jagpreetsinghsasan , apologies for not commenting here about the updates.
Currently here are the steps that I have done:
- Testing with container CLI opened. Verify by starting a flow and checking the flow status with the flow ID
- Compare result with the automatic testing code
- Recreate result with the corda demo repository and compare similarities and differences in coding in cactus repository
- Run postman to check connection with in corda v5

After the postman testing, we tried disabling the SSL certification verification there and the output was the expected successful start flow. Without it being disabled, the output is "certificate has expired" error. Currently trying to learn how to disable it in the repository in the axios part of the coding for corda v5 and will test it again once it is successful.

**Comment by johnhomantaring** [comment]
Hello @petermetz and @jagpreetsinghsasan can you assign this to me? We just have some ticket reassignment on our end and I will continue what we have done on this issue.

**Comment by jagpreetsinghsasan** [comment]
Hi @johnhomantaring, any updates on this? (Corda 5 support seems to be the current highest priority task in Cacti)

**Comment by jagpreetsinghsasan** [comment]
Hi @johnhomantaring Corda has now released the general available Corda v5.0
I have already created the AIO for the newest release (just need to add some healthcheck etc.) and will share the same here for usage for the connector soon. Please let me know on how can I contribute further towards its completion so that we can have this connector PR soon (very high priority task).

**Comment by jagpreetsinghsasan** [comment]
> Hi @johnhomantaring Corda has now released the general available Corda v5.0 I have already created the AIO for the newest release (just need to add some healthcheck etc.) and will share the same here for usage for the connector soon. Please let me know on how can I contribute further towards its completion so that we can have this connector PR soon (very high priority task).

Here's the AIO for the latest Corda v5.0: https://github.com/jagpreetsinghsasan/cactus/commit/72ef23586b6f476ac7d8333b92a43ce13b136e8b

**Comment by johnhomantaring** [comment]
> > Hi @johnhomantaring Corda has now released the general available Corda v5.0 I have already created the AIO for the newest release (just need to add some healthcheck etc.) and will share the same here for usage for the connector soon. Please let me know on how can I contribute further towards its completion so that we can have this connector PR soon (very high priority task).
> 
> Here's the AIO for the latest Corda v5.0: [jagpreetsinghsasan@72ef235](https://github.com/jagpreetsinghsasan/cactus/commit/72ef23586b6f476ac7d8333b92a43ce13b136e8b)

Thanks for this @jagpreetsinghsasan we also tried to move to Corda v5 because of the latest release but got some issues on postgres this will be a great help on the continuation of testing.

**Comment by jagpreetsinghsasan** [comment]
> > > Hi @johnhomantaring Corda has now released the general available Corda v5.0 I have already created the AIO for the newest release (just need to add some healthcheck etc.) and will share the same here for usage for the connector soon. Please let me know on how can I contribute further towards its completion so that we can have this connector PR soon (very high priority task).
> > 
> > 
> > Here's the AIO for the latest Corda v5.0: [jagpreetsinghsasan@72ef235](https://github.com/jagpreetsinghsasan/cactus/commit/72ef23586b6f476ac7d8333b92a43ce13b136e8b)
> 
> Thanks for this @jagpreetsinghsasan we also tried to move to Corda v5 because of the latest release but got some issues on postgres this will be a great help on the continuation of testing.

Even I faced that problem, couldn't solve with any of the prominent dind images out there. So for the time being, I used the host network for the containers to spin up (have added that in the readme). But that's what I will look into, maybe we can tweak the test-network configurations from the CDSE repo before actually spinning up the postgres.

**Comment by johnhomantaring** [comment]
> > > > Hi @johnhomantaring Corda has now released the general available Corda v5.0 I have already created the AIO for the newest release (just need to add some healthcheck etc.) and will share the same here for usage for the connector soon. Please let me know on how can I contribute further towards its completion so that we can have this connector PR soon (very high priority task).
> > > 
> > > 
> > > Here's the AIO for the latest Corda v5.0: [jagpreetsinghsasan@72ef235](https://github.com/jagpreetsinghsasan/cactus/commit/72ef23586b6f476ac7d8333b92a43ce13b136e8b)
> > 
> > 
> > Thanks for this @jagpreetsinghsasan we also tried to move to Corda v5 because of the latest release but got some issues on postgres this will be a great help on the continuation of testing.
> 
> Even I faced that problem, couldn't solve with any of the prominent dind images out there. So for the time being, I used the host network for the containers to spin up (have added that in the readme). But that's what I will look into, maybe we can tweak the test-network configurations from the CDSE repo before actually spinning up the postgres.

As of now I am still experiencing issue with the latest AIO using Mac (Intel), it seems that port 8888 is not accessible. On the other hand testing it with the team using Windows works perfectly. Checking if there are some changes we could work on to run it also on Mac.

**Comment by johnhomantaring** [comment]
> 

This is also using your latest changes with R3Corda V5. Already fixed the issue it seems that the firewall for ports on Mac is hindering the exposed ports 8888. I was able to fix the issue by bypassing the firewall https://github.com/docker/for-mac/issues/2670. Not sure if it's because of the limitation of the corporate laptop firewall rules or a Mac firewall issue but changing the proxyport resolves the issue.

Also tried using docker:20.10.3-dind and was able to successfully initiate the R3 Corda V5 AIO proceeding with the  changes on PluginLedgerConnectorCorda.

**Comment by aldousalvarez** [comment]
Hello @jagpreetsinghsasan I am helping on the testing of this issue.

**Comment by adrianbatuto** [comment]
The ticket is currently blocked by the issue
https://github.com/corda/CSDE-cordapp-template-kotlin/issues/120
https://github.com/corda/corda-runtime-os/issues/5284

**Comment by jagpreetsinghsasan** [comment]
Closing the issue as 5.0 assets are dropped in favour for 5.1 (and tracking it from scratch is easier than to push more time spent  in this same long ongoing task)
The followup task: [feat(corda): support 5.1 via TS/HTTP (no JVM)](https://github.com/hyperledger/cacti/issues/2978)
